Client/Frontend/Browser/SwipeAnimator.swift

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+struct SwipeAnimatorUX {
+    var totalRotationInDegrees: Double
+    var deleteThreshold: CGFloat
+    var totalScale: CGFloat
+    var totalAlpha: CGFloat
+    var minExitVelocity: CGFloat
+    var recenterAnimationDuration: NSTimeInterval
+}
+
+private let DefaultSwipeAnimatorUX =
+    SwipeAnimatorUX(
+        totalRotationInDegrees: 10,
+        deleteThreshold: 80,
+        totalScale: 0.9,
+        totalAlpha: 0,
+        minExitVelocity: 800,
+        recenterAnimationDuration: 0.15)
+
+protocol SwipeAnimatorDelegate: class {
+    func swipeAnimator(animator: SwipeAnimator, viewDidExitContainerBounds: UIView)
+}
+
+class SwipeAnimator: NSObject {
+    weak var delegate: SwipeAnimatorDelegate?
+
+    private var prevOffset: CGPoint!
+
+    let container: UIView
+    let animatingView: UIView
+    private let ux: SwipeAnimatorUX
+
+    var containerCenter: CGPoint {
+        return CGPoint(x: CGRectGetWidth(container.frame) / 2, y: CGRectGetHeight(container.frame) / 2)
+    }
+
+    init(animatingView: UIView, container: UIView, ux: SwipeAnimatorUX = DefaultSwipeAnimatorUX) {
```

Comment:
It's also strange that we're passing the `SwipeAnimatorUX` into the `init` here. I think we made these `UX` `struct`s just as a way to collect constants into a single place, with maybe the eventual goal that they'll be going into some shared constants file.

I guess you're treating `SwipeAnimatorUX` as a kind of skin?